### PR TITLE
Info Boxes: Group boxes by plugin, sort by priority

### PR DESCRIPTION
**Pull request by sethtroll**
Group InfoBoxes by plugin, then sorting those groups by priority

Boost Indicators are set as HIGH priority
Timers InfoBoxes are set as MED priority

Relevant to #559

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/infobox/InfoBoxOverlay.java [hunk 1]
```diff
@@ -62,13 +65,21 @@ public InfoBoxOverlay(InfoBoxManager infoboxManager, TooltipManager tooltipManag
...
-		List<InfoBox> infoBoxes = infoboxManager.getInfoBoxes();
-
-		if (infoBoxes.isEmpty())
+		if (infoboxManager.getInfoBoxes().isEmpty())
 		{
 			return null;
 		}
 
+		// Group by plugin
+		List<InfoBox> infoBoxes = infoboxManager.getInfoBoxes()
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/infobox/InfoBoxOverlay.java]: This should be done only if the available infoboxes changes, and not per frame.

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/infobox/InfoBoxOverlay.java [hunk 2]
```diff
@@ -72,8 +73,11 @@ public Dimension render(Graphics2D graphics, Point parent)
 		int width = infoBoxes.size() * (BOXSIZE + SEPARATOR);
 		int x = 0;
 
-		for (InfoBox box : infoBoxes)
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/infobox/InfoBoxOverlay.java]: Why was this changed?

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/infobox/InfoBox.java [hunk 3]
```diff
@@ -26,23 +26,35 @@
...
+		this.plugin = plugin;
+		setPriority(InfoBoxPriority.NONE);
 	}
 
 	public BufferedImage getImage()
 	{
 		return image;
 	}
 
+	public Plugin getPlugin()
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/infobox/InfoBox.java]: Use Lombok features

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/infobox/InfoBox.java [hunk 4]
```diff
@@ -26,23 +26,35 @@
 
 import java.awt.Color;
 import java.awt.image.BufferedImage;
+import net.runelite.client.plugins.Plugin;
 
 public abstract class InfoBox
 {
 	private final BufferedImage image;
 
+	private final Plugin plugin;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/infobox/InfoBox.java]: Instead of passing entire plugin here, pass only plugin's unique identifier, e.g PluginDescriptor

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/infobox/InfoBoxManager.java [hunk 5]
```diff
@@ -73,7 +83,26 @@ public void cull()
 			{
 				log.debug("Culling InfoBox {}", box);
 				it.remove();
+
+				refreshInfoBoxes();
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/infobox/InfoBoxManager.java]: This should probably be done once at the end, if an infobox was culled

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/infobox/InfoBoxManager.java [hunk 6]
```diff
@@ -73,7 +83,26 @@ public void cull()
...
 			}
 		}
 	}
+
+	private void refreshInfoBoxes()
+	{
+		// Group by plugin
+		List<InfoBox> sortedList = infoBoxes
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/infobox/InfoBoxManager.java]: https://stackoverflow.com/a/4258874/7189686

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/infobox/InfoBoxOverlay.java (hunk 2 above)
Comment (sethtroll) [reply runelite-client/src/main/java/net/runelite/client/ui/overlay/infobox/InfoBoxOverlay.java]: Oops, that should not have been added. Fixed.

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/infobox/InfoBoxManager.java [hunk 7]
```diff
@@ -25,37 +25,47 @@
...
 import lombok.extern.slf4j.Slf4j;
+import net.runelite.client.plugins.PluginDescriptor;
 
 @Singleton
 @Slf4j
 public class InfoBoxManager
 {
 	private final List<InfoBox> infoBoxes = new ArrayList<>();
 
+	private Boolean culled = Boolean.FALSE;
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/infobox/InfoBoxManager.java]: Why is this a member field?
